### PR TITLE
Xenobiology Control Console

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
A free look/advanced camera console that  can pick up and place slimes remotely, as well as deploy monkeys from stored monkey cubes. One of the biggest complaints about xenobio is how tedious and clunky it is dragging all the monkeys and slimes about, so hopefully this makes it more enjoyable.

The eye can only move inside the Xenobiology Lab so you can't use it to hunt badguys or dump slimes everywhere. However renaming areas to Xenobiology Lab will allow you to access them if you want to steal the blueprints and expand your territory.

DONE:
-Basically the whole machine. All the functions of it work in testing.
-Restricted to xenobiology only

TODO: 

-Icons
-Changelog
-Put it on the map(s)
-Make the eye autojump to xenobio on new()
-Clean up some copypaste in the CameraOff part

Thanks to @AnturK for making these advanced camera machines in the first place

### CODE REVIEW

**Comment by Incoming5643** [comment]
kinda takes all the danger out of it

**Comment by KorPhaeron** [comment]
@Incoming5643 

There isn't really any danger anyway. Once you feed the first two slimes they're your friends forever and each generation after that.

**Comment by TechnoAlchemisto** [comment]
So can you kill and process slimes through this console?

**Comment by KorPhaeron** [comment]
You could warp them into space just south to kill them but otherwise no, you still do that by hand.

**Comment by AnturK** [comment]
camerachunks don't track networks in any way so it's not really feasible without modifying them a bit.

**Comment by KorPhaeron** [comment]
How hard would that be to change?

**Comment by AnturK** [comment]
Pretty hard without big performance hit probably, ill have a look at it later.

**Comment by Giacom** [comment]
It wouldn't be hard but maybe tedious to test, you would need to add another layer to the system.

```
cameranet 1 --- ∞ cameranetwork 1 --- ∞ camerachunk
```

* You would have to add a new datum in between the cameranet and its list of chunks representing all the camera networks.
* It would then hold all the chunks and have all the information, such as class fields, it needs which cameranet is holding.
* Then when you add/remove/update a chunk, you would need to loop through the list and update each network with the data you already collected, such as what the camera can and cannot see.
* Finally you filter out chunks which cameras, within the network, can't see.

Possible problems:
* If you were wondering about duplicate chunks then you would maybe want to optimise it so that different networks are referencing the same chunk but memory size if not a big bottleneck in BYOND so having duplicate chunks in each network is not the end of the world. The optimisation would also be a huge pain in the ass to make sure it works correctly.
* Filtering out the cameras, which aren't in the network, would probably involve looping through cameras and that will probably add some lag to the system already.

**Comment by KorPhaeron** [comment]
If it's gonna be a performance issue it might make the most sense to make a new type of eye that cant move outside of xenobio

**Comment by GunHog** [comment]
Does the RD get one of these for his office?  Oh, and very important, if you are NOT giving one of these to the RD but you are putting it on the RD network, make sure it cannot view the AI's upload or chamber.

**Comment by KorPhaeron** [comment]
@GunHog 

I think it'd be incredibly annoying if a guy in another room could remotely mess up your workplace.

I'm gonna redo it so the camera eye can only be in xenobio period anyway.

**Comment by Midaychi** [comment]
A suggestion, for this. Make the functionality tied to robotic arms, that are mounted in the cages. Slimes that are sufficiently pissed off can bash these robotic arms into submission, leading to their need for being fixed and replaced and maintained.

In the future, such a device could be mainlined to a buildable device for all sorts of devious station-wide purposes.

**Comment by KorPhaeron** [comment]
Made the commit messy will re-open
